### PR TITLE
[risk=no][RW-7973] A few more UI tests for async

### PR DESCRIPTION
**Pull request by jmthibault79**
---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

**Review by aweng98 — APPROVED** [review]
LGTM! one question only.

File: ui/src/app/pages/workspace/workspace-edit.spec.tsx [hunk 1]
```diff
@@ -687,6 +748,48 @@ describe('WorkspaceEdit', () => {
...
+    await waitOneTickAndUpdate(wrapper);
+
+    wrapper
+      .find('[data-test-id="workspace-confirm-save-btn"]')
+      .first()
+      .simulate('click');
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(workspacesApi.workspaceOperations.length).toEqual(numBefore + 1);
+    expect(mockNavigate).toHaveBeenCalledTimes(1);
```
Comment (evrii) [inline ui/src/app/pages/workspace/workspace-edit.spec.tsx:790]: Is there an error or warning message if a user tries to create more than one workspace?

**Review by evrii — APPROVED** [review]

File: ui/src/app/pages/workspace/workspace-edit.spec.tsx (hunk 1 above)
Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-edit.spec.tsx:790]: the fix was to make it no longer possible to do so: the button is disabled after clicking

File: ui/src/app/pages/workspace/workspace-edit.spec.tsx [hunk 2]
```diff
@@ -411,13 +411,72 @@ describe('WorkspaceEdit', () => {
...
+    wrapper
+      .find('[data-test-id="workspace-confirm-save-btn"]')
+      .first()
+      .simulate('click');
+    await waitOneTickAndUpdate(wrapper);
+    expect(workspacesApi.workspaceOperations.length).toEqual(numBefore + 1);
+    expect(mockNavigate).toHaveBeenCalledTimes(1);
+  });
+
+  it('defaults to upgrading the CDR Version when synchronously duplicating a workspace with an older CDR Version', async () => {
```
Comment (aweng98) [inline ui/src/app/pages/workspace/workspace-edit.spec.tsx:442]: Is this test and next one are essential same but with different value in flag `environment.enableAsyncWorkspaceOperations`? If so, I recommend using `it.each`.

Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-edit.spec.tsx:442]: not completely - one counts `workspaces` and the other `workspaceOperations`

**Pull request merged**